File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        i = 0
        for num in nums:
            if num in seen.keys():
                seen[num].append(i)
            else:
                seen[num] = [i]
            i += 1
        
        i = 0
        while (i < len(nums) - 1):
            cur = target - nums[i]
            if cur in seen.keys():
                if cur == nums[i] and len(seen[cur]) > 1:
                    return [seen[cur][0], seen[cur][1]]
                elif cur != nums[i]:
                    return [i, seen[cur][0]]
            
            i += 1

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (seen.has(complement)) {
            return [seen.get(complement), i];
        }
        
        seen.set(nums[i], i);
        
        // if (seen.has(nums[i])) {
        //     seen.get(nums[i]).push(i);
        // } else
        //     seen.set(nums[i], [i]);
    }
    
};